File: sbe/binance/spot_3_2/generated-php/spot_sbe/WebSocketSessionLogonResponse.php
<?php
/**
 * Generated SBE (Simple Binary Encoding) message codec.
 */

class WebSocketSessionLogonResponse
{
    public const TEMPLATE_ID = 51;
    public const SCHEMA_ID = 3;
    public const SCHEMA_VERSION = 2;
    public const BLOCK_LENGTH = 26;

    public int|float|array|null $authorizedSince = null;
    public int|float|array|null $connectedSince = null;
    public int|float|array|null $returnRateLimits = null;
    public int|float|array|null $serverTime = null;
    public int|float|array|null $userDataStream = null;
    public string $loggedOnApiKey = '';

    private function decodeVarData(string $data, int &$offset): string
    {
        $length = unpack('V', substr($data, $offset, 4))[1];
        $offset += 4;
        $value = substr($data, $offset, $length);
        $offset += $length;
        return $value;
    }

    public function encode(): string
    {
        $buffer = '';

        if ($this->authorizedSince !== null) {
            $buffer .= pack('q', $this->authorizedSince);
        }
        if ($this->connectedSince !== null) {
            $buffer .= pack('q', $this->connectedSince);
        }
        if ($this->serverTime !== null) {
            $buffer .= pack('q', $this->serverTime);
        }

        return $buffer;
    }

    public function decode(string $data): void
    {
        $offset = 0;

        $this->authorizedSince = unpack('q', substr($data, $offset, 8))[1];
        $offset += 8;
        $this->connectedSince = unpack('q', substr($data, $offset, 8))[1];
        $offset += 8;
        $this->serverTime = unpack('q', substr($data, $offset, 8))[1];
        $offset += 8;

        // Skip to end of block for forward compatibility
        $offset = 26;


        $this->loggedOnApiKey = $this->decodeVarData($data, $offset);
    }
}

File: sbe/binance/spot_3_1/generated-python/spot_sbe/WebSocketSessionLogonResponse.py
"""Generated SBE (Simple Binary Encoding) message codec."""

import struct
from typing import List, Optional, Tuple, Set
from io import BytesIO

class WebSocketSessionLogonResponse:
    """SBE message: WebSocketSessionLogonResponse."""

    TEMPLATE_ID = 51
    SCHEMA_ID = 3
    SCHEMA_VERSION = 1
    BLOCK_LENGTH = 26

    def __init__(self):
        self.authorized_since = None
        self.connected_since = None
        self.return_rate_limits = None
        self.server_time = None
        self.user_data_stream = None
        self.logged_on_api_key = b''

    def _decode_var_data(self, data: bytes, offset: int) -> Tuple[bytes, int]:
        """Decode variable-length binary data."""
        pos = offset
        length = struct.unpack_from('<I', data, pos)[0]
        pos += 4
        value = data[pos:pos+length]
        pos += length
        return (value, pos)

    def encode(self) -> bytes:
        """Encode the message to bytes."""
        buffer = BytesIO()

        if self.authorized_since is not None:
            buffer.write(struct.pack('<q', self.authorized_since))
        if self.connected_since is not None:
            buffer.write(struct.pack('<q', self.connected_since))
        if self.server_time is not None:
            buffer.write(struct.pack('<q', self.server_time))

        return buffer.getvalue()

    def decode(self, data: bytes) -> None:
        """Decode the message from bytes."""
        pos = 0

        self.authorized_since = struct.unpack_from('<q', data, pos)[0]
        pos += 8
        self.connected_since = struct.unpack_from('<q', data, pos)[0]
        pos += 8
        self.server_time = struct.unpack_from('<q', data, pos)[0]
        pos += 8

        # Skip to end of block for forward compatibility
        pos = 26


        self.logged_on_api_key, pos = self._decode_var_data(data, pos)
